File: hw/chicaiza/u3/hw27GuisOperations/SchoolSystemJs/ec/edu/espe/schoolsystem/utils/MongoDBConnection.js
const mongoose = require('mongoose');

class MongoDBConnection {
    static async connect() {
        try {
            await mongoose.connect(
                'mongodb+srv://daniel:daniel2007@cluster0.v7buh9x.mongodb.net/school?appName=Cluster0'
            );
            console.log('Connected to MongoDB Atlas (database: school)');
        } catch (err) {
            console.error('Error connecting to MongoDB:', err);
            throw err; // para que el controlador sepa si falla
        }
    }

    static close() {
        mongoose.connection.close();
        console.log('MongoDB connection closed');
    }
}

module.exports = MongoDBConnection;

File: hw/chicaiza/u3/hw27GuisOperations/FitnessGym/src/main/java/utils/MongoDBConnection.java
package utils;
import com.mongodb.client.MongoDatabase;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;

/**
 *
 * @author Daniel
 */
public class MongoDBConnection {
    
    private static MongoDatabase database;

    
    public static MongoDatabase getDatabase() {
        
        if (database == null) {
            
            MongoClient client = MongoClients.create("mongodb+srv://daniel:daniel2007@cluster0.v7buh9x.mongodb.net/?appName=Cluster0");
            database = client.getDatabase("FitnesGym");
        }
        
        return database;
    }
    
}
